Cabal/Distribution/Backpack/Id.hs

```diff
@@ -0,0 +1,198 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE PatternGuards #-}
+-- | See <https://github.com/ezyang/ghc-proposals/blob/backpack/proposals/0000-backpack.rst>
+module Distribution.Backpack.Id(
+    computeComponentId,
+    computeCompatPackageKey,
+    computeCompatPackageName,
+) where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import Distribution.Simple.Compiler hiding (Flag)
+import Distribution.Package
+import Distribution.PackageDescription as PD hiding (Flag)
+import Distribution.Simple.Setup as Setup
+import qualified Distribution.Simple.InstallDirs as InstallDirs
+import Distribution.Simple.LocalBuildInfo
+import Distribution.Utils.Base62
+import Distribution.Version
+
+import Distribution.Text
+    ( display, simpleParse )
+
+-- TODO: move this to cabal-install
```

Comment:
Does this TODO make sense given the comment below about this being for "good enough" component ids. We need a default if `--ipid` isn't specified, so this does have to stay here no? Presumably it's used in the "Simple" build system.